File: 프로그래머스/lv1/118666. 성격 유형 검사하기/성격 유형 검사하기.java
import java.util.*;
class Solution {
    public String solution(String[] survey, int[] choices) {
        String answer = "";
        HashMap<String, Integer> result = new HashMap<>();
        String[] mbti = {"R","T","C","F","J","M","A","N"};
        
        for(String m: mbti)
            result.put(m,0);
        
        for(int i=0;i<choices.length;i++){
            String l = survey[i].substring(0,1);
            String r = survey[i].substring(1,2);
            if(choices[i]<=3){
                result.put(l,result.get(l)+(4-choices[i]));
            }
            else if(choices[i]>=5){
                result.put(r,result.get(r)+choices[i]-4);
            }
        }
        
        answer+=(result.get("R")>=result.get("T"))?"R":"T";
        answer+=(result.get("C")>=result.get("F"))?"C":"F";
        answer+=(result.get("J")>=result.get("M"))?"J":"M";
        answer+=(result.get("A")>=result.get("N"))?"A":"N";
        
        return answer;
    }
}

File: 프로그래머스/1/118666. 성격 유형 검사하기/성격 유형 검사하기.py
def solution(survey, choices):
    answer = ''
    scores={'R':0,'T':0,'C':0,'F':0,'J':0,'M':0,'A':0,'N':0}
    
    for i in range(len(choices)):
        if choices[i]>=5:
            scores[survey[i][1]]+=choices[i]-4
        elif choices[i]<=3:
            scores[survey[i][0]]+=4-choices[i]
    
    personal=list(scores)
    
    for i in range(0,len(personal),2):
        answer+=personal[i] if scores[personal[i]]>=scores[personal[i+1]] else personal[i+1]

    return answer